cmd/cosign/cli/verify_manifest.go

```diff
@@ -0,0 +1,135 @@
+	cmd := VerifyManifestCommand{VerifyCommand: VerifyCommand{}}
+	flagset := flag.NewFlagSet("cosign verify-manifest", flag.ExitOnError)
+	applyVerifyFlags(&cmd.VerifyCommand, flagset)
+
+	return &ffcli.Command{
+		Name:       "verify-manifest",
+		ShortUsage: "cosign verify-manifest -key <key path>|<key url>|<kms uri> <path/to/manifest>",
+		ShortHelp:  "Verify all signatures of images specified in the manifest",
+		LongHelp: `Verify all signature of images in a Kubernetes resource manifest by checking claims
+against the transparency log.
+
+Shell-like variables in the Dockerfile's FROM lines will be substituted with values from the OS ENV.
+
+EXAMPLES
+  # verify cosign claims and signing certificates on images in the manifest
+  cosign verify-manifest <path/to/my-deployment.yaml>
+
+  # additionally verify specified annotations
+  cosign verify-manifest -a key1=val1 -a key2=val2 <path/to/my-deployment.yaml>
+
+  # (experimental) additionally, verify with the transparency log
+  COSIGN_EXPERIMENTAL=1 cosign verify-dockerfile <path/to/my-deployment.yaml>
+
+  # verify images with public key
+  cosign verify-manifest -key cosign.pub <path/to/my-deployment.yaml>
+
+  # verify images with public key provided by URL
+  cosign verify-manifest -key https://host.for/<FILE> <path/to/my-deployment.yaml>
+
+  # verify images with public key stored in Azure Key Vault
+  cosign verify-manifest -key azurekms://[VAULT_NAME][VAULT_URI]/[KEY] <path/to/my-deployment.yaml>
+
+  # verify images with public key stored in AWS KMS
+  cosign verify-manifest -key awskms://[ENDPOINT]/[ID/ALIAS/ARN] <path/to/my-deployment.yaml>
+
+  # verify images with public key stored in Google Cloud KMS
+  cosign verify-manifest -key gcpkms://projects/[PROJECT]/locations/global/keyRings/[KEYRING]/cryptoKeys/[KEY] <path/to/my-deployment.yaml>
+
+  # verify images with public key stored in Hashicorp Vault
+  cosign verify-manifest -key hashivault://[KEY] <path/to/my-deployment.yaml>`,
+
+		FlagSet: flagset,
+		Exec:    cmd.Exec,
+	}
+}
+
+// Exec runs the verification command
+func (c *VerifyManifestCommand) Exec(ctx context.Context, args []string) error {
+	if len(args) != 1 {
+		return flag.ErrHelp
+	}
+
+	manifestPath := args[0]
+
+	err := isExtensionAllowed(manifestPath)
+	if err != nil {
+		return errors.Wrap(err, "check if extension is valid")
+	}
+
+	manifest, err := ioutil.ReadFile(manifestPath)
+	if err != nil {
+		return fmt.Errorf("could not read manifest: %v", err)
+	}
+
+	images, err := getImagesFromYamlManifest(string(manifest))
+	if err != nil {
+		return fmt.Errorf("failed extracting images from manifest: %v", err)
+	}
+	if len(images) == 0 {
+		return errors.New("no images found in manifest")
+	}
+	fmt.Fprintf(os.Stderr, "Extracted image(s): %s\n", strings.Join(images, ", "))
+
+	return c.VerifyCommand.Exec(ctx, images)
+}
+
+func getImagesFromYamlManifest(manifest string) ([]string, error) {
+	var images []string
+	re := regexp.MustCompile(`image:\s?(?P<Image>.*)\s?`)
```

Comment:
suggestion: Perhaps it could be more robust to decode the manifest and validate that the extensions are among those identified in the issue (Pod, DaemonSet...), then use the decoded object to get the image value for all the containers (e.g. `obj.Spec.Containers[*].Image`). 
Otherwise you could have a some resources that might not be among the identified list, e.g. CRs.

Reply:
Let me know if you want me to help you with this.

Reply:
@hectorj2f yes, this was a consideration as mentioned in https://github.com/sigstore/cosign/issues/437#issuecomment-887084109 - I think it would be better, but as different Kinds come out, it binds us more to the K8s API for parsing unless we still just walk down the tree looking for "image" keys (but guaranteed valid YAML) which would address your other comment.

This regex also doesn't account for quoted string variants, so either way this naive approach needs to change in _some_ fashion.

Reply:
Agreed. There's prior art for ducktyping `PodSpec`-like Objects, but I don't think it's worth blocking merge for a "perfect solution

Reply:
> There's prior art for ducktyping PodSpec-like Objects. 

Yes, you could use runtime.Raw and decode the object to the list of supported types here.

Sure, the current implementation should not be blocked by this. Once this PR is merged, I could open a PR with my proposed changes, so you can have a look at them.